chandra_aca/fov_plot.py

```diff
@@ -0,0 +1,309 @@
+    return fig
+
+
+def bad_acq_stars(stars):
+    """
+    Return mask of 'bad' stars, by evaluating AGASC star parameters.
+    :param stars: astropy table-compatible set of agasc records of stars
+    :returns: boolean mask true for 'bad' stars
+    """
+    return ((stars['CLASS'] != 0) |
+            (stars['MAG_ACA_ERR'] > 100) |
+            (stars['POS_ERR'] > 3000) |
+            (stars['ASPQ1'] > 0) |
+            (stars['ASPQ2'] > 0) |
+            (stars['ASPQ3'] > 999) |
+            (stars['VAR'] > -9999))
+
+
+def plot_catalog(ra, dec, roll, catalog, starcat_time=DateTime(),
+                 stars=None, bad_stars=None, red_mag_lim=None, title=None):
+    """
+    Make standard catalog plot with a star field and the elements of a catalog.
+
+    :param ra: RA in degrees
+    :param dec: Dec in degrees
+    :param roll: Roll in degrees
+    :param catalog: list of dicts or other astropy.table compatible record structure with
+                    conventional catalog parameters for a set of
+                    ACQ/BOT/GUI/FID/MON items.
+    :param starcat_time: star catalog time.  Used as time for proper motion correction.
+    :param stars: astropy table compatible set of agasc records of stars.  If not supplied,
+                  these will be fetched for the supplied attitude
+    :param bad_stars: mask of stars that should be plotted in BAD_STAR_COLOR
+    :param title: string to be used as suptitle for the figure
+    :returns: matplotlib figure
+    """
+    if stars is None:
+        stars = agasc.get_agasc_cone(ra, dec,
+                                     radius=1.5,
+                                     date=DateTime(starcat_time).date)
+    fig = star_plot(catalog, attitude=[ra, dec, roll], stars=stars, title=title,
+                    bad_stars=bad_stars, red_mag_lim=red_mag_lim)
+    return fig
+
+
+def plot_star_field(ra, dec, roll, starcat_time=DateTime(), stars=None, bad_stars=None,
+                    red_mag_lim=None, title=None):
+    """
+    Make standard star field plot.
+
+    :param ra: RA in degrees
+    :param dec: Dec in degrees
+    :param roll: Roll in degrees
+    :param starcat_time: star catalog time.  Used as time for proper motion correction.
+    :param stars: astropy table compatible set of agasc records of stars.  If not supplied,
+                  these will be fetched for the supplied attitude
+    :param bad_stars: mask of stars that should be plotted in BAD_STAR_COLOR
+    :param title: string to be used as suptitle for the figure
+    :returns: matplotlib figure
+    """
+    if stars is None:
+        stars = agasc.get_agasc_cone(ra, dec,
+                                     radius=1.5,
+                                     date=DateTime(starcat_time).date)
+    fig = star_plot(catalog=None, attitude=[ra, dec, roll], stars=stars, title=title,
+                    quad_bound=False, red_mag_lim=red_mag_lim, bad_stars=bad_stars)
+    return fig
+
+
+def plot_compass(roll):
+    """
```

Comment:
Not sure if I should leave this here or put it back in starcheck.  It seems pretty general.